Automattic/jetpack
# Update/markdown gutenberg block improvements

### Changes proposed in this Pull Request:

#### Description

Adds some improvements over the original Markdown Block (PR #9705):

1. [ ] Basic text editing buttons to make text bold and italic, create links and add images. They should mimic the UX in the RichText Gutenberg editor component.
2. [ ] Add a checkbox to allow the user disable the live-preview (for very big markdown sources).
3. [ ] A Markdown syntax legend in the block description (for Markdown newbies).
4. [ ] Fix caret positioning after intro in Internet Explorer 11.
5. [ ] Add tests for the live preview component
6. [ ] Add tests for the markdown renderer component
7. [ ] Add tests for the markdown converter library
8. [ ] Add tests for the caret management library

This PR notes will be updated as it evolves.

### Testing instructions:

Steps to test the Markdown block component:

1. Install the Gutenberg plugin in your WordPress environment.
2. Enable "Write posts or pages in plain-text Markdown syntax" in your Jetpack config page.
3. Create a new post, add a Markdown block and write some markdown code.
4. Save a draft of the post, or publish it.
5. The markdown code should be rendered as HTML in the published post/draft.

Posts can be edited as well.

### Proposed changelog entry for your changes:

Adds formatting controls and custom settings to the Markdown Block.

### Checklist:
